Use the accessor names as written in Objective-C.

varungandhi-apple opened the pull request:
Not 100% sure if my approach taken here is correct. It is unclear to me what the semantics of `getParsedAccessor` are supposed to be in the presence of user-written accessor names in Objective-C (as opposed to those written in Swift). Should they be passed through? Or not passed through because they are considered implicit? I've assumed that the existing behavior of `getParsedAccessor` is correct, so we use `getAccessor` instead.

Fixes rdar://problem/55519276.

Related: commit 0c5d52d8609aaa3016621887d03571c7582984a5 (PR #26461) where `getParsedAccessor` was introduced.

varungandhi-apple committed:
Add minimal test case for rdar://problem/55519276.

We are dropping the getter and setter names we get from Objective-C.
This leads to a runtime crash. Oops.

varungandhi-apple commented:
@swift-ci please test

Azoy commented on test/PrintAsObjC/getter_setter.swift:
```diff
@@ -0,0 +1,15 @@
+  public var number: Int64
+  // CHECK: @property (nonatomic, getter=getPower, setter=setPower:) int64_t number;
+  public override init() {
+    number = 9001
```
(It’s over 9000!?!)

varungandhi-apple committed:
[AST] Get the ObjC accessor name, not the one parsed from Swift.

Fixes rdar://problem/55519276.

varungandhi-apple commented:
@swift-ci please test

rjmccall commented:
IIRC, Slava was aiming to eliminate uses of the raw `getAccessor` method.  I think `getParsedAccessor` is meant to be solely about what was written in source, so it's reasonable to say that we shouldn't be using it in this semantic method, but it's not clear to me that there's currently a better method, since we (apparently) can't assert as a precondition that this is an `@objc` property.  Slava would have thoughts.

slavapestov commented:
I think the right fix is to not mark imported accessors that correspond to objc methods as implicit. Only bona fide synthesized accessors should be implicit (eg, those for stored properties of structs, or a modify accessor synthesized when a property witnesses a mutable protocol requirement, etc).

varungandhi-apple commented:
@swift-ci please test Linux platform

varungandhi-apple committed:
[NFC] Replace new getAccessor calls with getOpaqueAccessor calls.

It is fine if we synthesize the accessors a bit more than usual as
PrintAsObjC isn't performance-sensitive and IRGen needs the accessors
anyways.

varungandhi-apple commented:
@swift-ci please test

varungandhi-apple commented:
Note: I've added a new commit as I think @edymtt cherry-picked the first two commits elsewhere, so I didn't want to mess with history by squashing the third commit on top of the second one, just in case it causes issues.

varungandhi-apple commented:
Note for anyone else reading: the imported accessor is explicit in the existing code. Which is correct behavior. Hence, I didn't make any changes there.
